contents/javascript/Javascript-framework-VueJs.md

```diff
@@ -228,6 +234,21 @@ In this case, only the `root` component can be accessed in Vue while the rest ar
 
 <br>
 
+7. **Lifecycle Hooks**<br>
+    Every Vue instance goes through a series of initialization steps when it is created, i.e. setting up data observation, compiling the template, mounting the instance to the DOM, and updating the DOM when data modifies. Along these steps, Vue runs functions called lifecycle hooks, allowing users to add code at specific steps.<br/><br/>
+    For example, the `created` hook can be used to run code after the Vue instance is created:
+    ```js
+    Vue.component('todo-list', {
+      ...
+      created: function() {
+        console.log("to-do list is created.");
+      }
+    })
+    ```
+   There are also other hooks that will be called at different stages of the instance's lifecycle, such as `beforeCreate`, `beforeMount`, `mounted`, `beforeUpdate`, `updated`, `beforeDestroy` and `destroyed`. The lifecycle diagram below shows specific stages when each hook is executed:<br/>
```

Comment:
Perhaps can restructure your explanation to support the diagram below?

Reply:
Thanks for the comments! I have updated the chapter based on your comments. 

I also restructured the explanation for the lifecycle hooks section.